utils/gencorp.py

```diff
@@ -0,0 +1,222 @@
+basepath, _ = os.path.split(os.path.realpath(__file__))
+_UTILS = os.sep + "utils"
+if basepath.endswith(_UTILS):
+    basepath = basepath[0 : -len(_UTILS)]
+    sys.path.append(basepath)
+
+from settings import Settings
+from article import Article
+from tree import Tree
+
+# To make this work, clone Miðeind Annotald repo, enter the Greynir
+# virtualenv and run "python setup.py develop" from the Annotald repo root
+from annotald.reynir_utils import reynir_sentence_to_annotree, simpleTree2NLTK
+from annotald.annotree import AnnoTree
+
+
+# Min num tokens in sentence
+MIN_SENT_LENGTH = 3
+
+# Num sentences per batch to shuffle
+MAX_BATCH = 10000
+
+# Separator for sentence trees in output file
+SEPARATOR = "\n\n"
+
+# Skip sentences containing these tokens
+ENGLISH_WORDS = frozenset(
+    [
+        "the",
+        "a",
+        "is",
+        "each",
+        "year",
+        "our",
+        "on",
+        "in",
+        "and",
+        "this",
+        "that",
+        "they",
+        "what",
+        "when",
+        "s",
+        "t",
+        "don't",
+        "isn't",
+        "big",
+        "cheese",
+        "steak",
+        "email",
+        "search",
+        "please",
+    ]
+)
+
+
+def gen_simple_trees(criteria):
+    """ Generate simplified parse trees from articles matching the criteria """
+    for a in Article.articles(criteria):
+        # Skip articles from certain websites
+        if (
+            not a.root_domain
+            or "raduneyti" in a.root_domain
+            or "lemurinn" in a.root_domain
+        ):
+            continue
+
+        # Load tree from article
+        try:
+            tree = Tree(url=a.url, authority=a.authority)
+            tree.load(a.tree)
+        except Exception as e:
+            print("Exception loading tree in {0}: {1}".format(a.url, e))
+            continue
+
+        # Yield simple trees
+        for ix, stree in tree.simple_trees():
+            text = stree.text
+            tokens = text.split()
+            if len(tokens) >= MIN_SENT_LENGTH:
+                wordset = set([t.lower() for t in tokens])
+                # Only return sentences without our bag of English words
+                if not (wordset & ENGLISH_WORDS):
+                    yield stree, tree.score(ix), tree.length(ix), a.uuid, a.url, ix
+
+
+def main(num_sent, parse_date_gt, outfile):
+
+    try:
+        # Read configuration file
+        Settings.read(os.path.join(basepath, "config/GreynirSimple.conf"))
+    except ConfigError as e:
+        print("Configuration error: {0}".format(e))
+        quit()
+
+    # Generate the parse trees from visible roots only,
+    # in descending order by time of parse
+    criteria = {"order_by_parse": True, "visible": True}
+    if parse_date_gt is not None:
+        criteria["parse_date_gt"] = parse_date_gt
+
+    # Generator for articles
+    gen = gen_simple_trees(criteria)
+
+    accumulated = []
+    total = 0
+    skipped = 0
+
+    with open(outfile, "w") as f:
```

Comment:
Add `, encoding="utf-8"`  (encoding is otherwise platform dependent, and it's not utf-8 on Windows)